src/yarp.c

```diff
@@ -2001,18 +2001,26 @@ parse_statements(yp_parser_t *parser, yp_context_t context) {
 // Parse an individual argument.
 static yp_node_t *
 parse_argument(yp_parser_t *parser, yp_node_t *arguments) {
-  yp_node_t *argument;
-
   if (accept(parser, YP_TOKEN_DOT_DOT_DOT)) {
     if (!yp_token_list_includes(&parser->current_scope->as.scope.locals, &parser->previous)) {
       yp_error_list_append(&parser->error_list, "unexpected ... when parent method is not forwarding.", parser->previous.start - parser->start);
     }
-    argument = yp_node_forwarding_arguments_node_create(parser, &parser->previous);
-  } else {
-    argument = parse_expression(parser, BINDING_POWER_NONE, "Expected to be able to parse an argument.");
+    return yp_node_forwarding_arguments_node_create(parser, &parser->previous);
   }
 
-  return argument;
+  if (accept(parser, YP_TOKEN_STAR)) {
+    yp_token_t previous = parser->previous;
+    if (*parser->current.start == ',' || *parser->current.start == ')') {
```

Comment:
Should do the same but more idiomatic:
```suggestion
    if (accept_any(parser, 2, YP_TOKEN_COMMA, YP_TOKEN_PARENTHESIS_RIGHT) {
```